Provider/src/main/java/dev/openfeature/contrib/providers/ConfidenceFeatureProvider.kt

```diff
@@ -253,4 +272,38 @@ class ConfidenceFeatureProvider private constructor(
             )
         }
     }
-}
\ No newline at end of file
+}
+
+suspend fun ConfidenceFeatureProvider.awaitProviderReady(
+    dispatcher: CoroutineDispatcher = Dispatchers.IO
+) = suspendCancellableCoroutine { continuation ->
+    val coroutineScope = CoroutineScope(dispatcher)
+    coroutineScope.launch {
+        observeProviderReady()
+            .take(1)
+            .collect {
+                continuation.resumeWith(Result.success(Unit))
+            }
+    }
+
+    coroutineScope.launch {
+        EventHandler.eventsObserver()
```

Comment:
wouldn't this use the global (singleton) eventshandler instead of the one created in the `provider`?